doc/akri_connector/connector-metadata-schema.json

```diff
@@ -126,14 +126,45 @@
       },
       "additionalProperties": false,
       "required": ["singular", "plural"]
+    },
+    "fieldNecessity": {
+      "type": "object",
+      "description": "Details about if a field is required or not and what shape its value should take",
+      "properties": {
+        "input": {
+          "type": "string",
+          "description": "If this field is relevant to this connector or not. If required, the connector cannot function without this field being provided in the specified shape. If unsupported, the connector will ignore any value provided for this field. If optional, the connector can function with or without this field being provided, but the behavior may vary depending on it.",
+          "enum": [
+            "required",
+            "unsupported",
+            "optional"
+          ]
+        },
+        "regex": {
+          "type": "array",
+          "description": "The set of regex that this field's value should adhere to.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "exampleValue": {
+          "type": "string",
+          "description": "An example value for this field that fits the expected format"
+        },
+        "description": {
+          "type": "string",
+          "description": "Additional user-readable context for this field. May explain how the field is used or explain the validation rules around it."
+        }
+      },
+      "required": ["input"]
     }
   },
   "properties": {
     "aioConnectorMetadataSchemaVersion": {
       "type": "string",
       "description": "The version of AIO connector metadata schema that this connector metadata file adheres to.",
       "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9A-Za-z-][0-9A-Za-z-]*)(?:\\.(?:0|[1-9A-Za-z-][0-9A-Za-z-]*))*))?(?:\\+([0-9A-Za-z-]+(?:\\.[0-9A-Za-z-]+)*))?$",
-      "default": "3.0-preview"
+      "default": "4.0-preview"
```

Comment:
Just checking if there isn't something like `$ref: #/aioConnectorMetadataSchemaVersion` that can be used here

Reply:
Not that I kind find, unfortunately